### DISCUSSION TITLE
Text transform uppercase first letter only

### DISCUSSION THREAD

**Discussion by victorocna** [tailwindlabs/tailwindcss#5693] [Ideas]
I was writing some tutorials for my team on text transformation (with CSS only) and I noticed something missing. Currently Tailwind has `uppercase`, `lowercase` and `capitalize` classes [very well documented here](https://tailwindcss.com/docs/text-transform), but nothing that just transforms only the first letter to uppercase.

How about a new class like `uppercase-first` or `first-letter` to do just that? I currently have this custom CSS class in every project that "extends" Tailwind's text transformation utilities.

```css
.first-letter::first-letter {
  text-transform: uppercase;
}
```

Cheers!

**Comment by adamwathan** [comment]
Kinda like this? :)

https://play.tailwindcss.com/GVOsrIO8jN

**Comment by victorocna** [comment]
Exactly right! Is this an existing class in Tailwind? I searched the docs and found nothing.

And what a great class name, I wish I thought of that.

**Reply by adamwathan** [reply]
Yeah it's new to the JIT engine, you can do anything with the first-letter, so for example:

```
<div class="first-letter:text-red-500 first-letter:font-bold first-letter:uppercase">
```

https://tailwindcss.com/docs/just-in-time-mode#pseudo-element-variants

**Comment by juanjinario** [comment]
Why this combination `"line-clamp-2 first-letter:uppercase"` does not work?

**Comment by oldravian** [comment]
**first-letter:uppercase** is not working inside **span** tag
`<span class="first-letter:uppercase">hello there</span>`

**Reply by wongjn** [reply]
As per the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/CSS/::first-letter) (emphasis mine):

> The ::first-letter [CSS](https://developer.mozilla.org/en-US/docs/Web/CSS) [pseudo-element](https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-elements) applies styles to the first letter of the first line of a [**block container**](https://developer.mozilla.org/en-US/docs/Web/CSS/Visual_formatting_model#block_containers), but only when not preceded by other content (such as images or inline tables).

`<span>` elements are inline by default, hence not a block container and hence the `::first-letter` pseudo-element not working in this case. You could apply `display: block` via the `block` class as an example to make it a block container and thus get the `::first-letter` to work: https://play.tailwindcss.com/BpHgKu6ZQn

**Reply by kelvinsekx** [reply]
Or better still add an "inline-block" class

**Comment by Durga-Zoca** [comment]
First-letter:uppercase not working on safari
 className={`w-[90px] text-base lowercase first-letter:uppercase md:text-lg mid:text-base ${textColor} text-justify`}
